enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -137,6 +138,18 @@ func NewCOWStore(ctx context.Context, env environment.Env, chunks []*Mmap, chunk
 	return s, nil
 }
 
+func (c *COWStore) lock() {
+	_, span := tracing.StartSpan(c.ctx)
+	defer span.End()
+	c.mu.Lock()
+}
```

Comment:
I think @tylerwilliams recently enabled locking (/ blocking?) profiles for pprof that you can use to see how much time is spent locking. I imagine that tracing each individual lock might add a lot of overhead and yield noisy measurements

Reply:
Sounds good 👍 This was just for some personal testing. Wasn't planning on merging this